3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingEnabledPostProcessorTest.java

```diff
@@ -0,0 +1,237 @@
+        boolean isBeanPostProcessorPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getInterfaces())
+                .anyMatch(i -> i.isAssignableFrom(
+                        BeanPostProcessor.class));
+        assertTrue(isBeanPostProcessorPresent);
+    }
+
+    @Test
+    @Order(10)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor overrides postProcessAfterInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_overridesPostProcessAfterInitMethod() {
+        try {
+            TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethod(
+                    "postProcessAfterInitialization", Object.class, String.class);
+        } catch (NoSuchMethodException e) {
+            throw new AssertionError("Expected that method postProcessAfterInitialization is overridden in " +
+                    "TrimmedAnnotationBeanPostProcessor, but was not.");
+        }
+    }
+
+    @Test
+    @Order(11)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor does not override postProcessBeforeInitialization() method")
+    void trimmedAnnotationBeanPostProcessor_notOverridesPostProcessAfterInitMethod() {
+        boolean isBeforeProcessingMethodPresent = Arrays.stream(TrimmedAnnotationBeanPostProcessor.class.getDeclaredMethods())
+                .anyMatch(m -> m.getName().equals("postProcessBeforeInitialization"));
+
+        assertFalse(isBeforeProcessingMethodPresent);
+    }
+
+    @Test
+    @Order(12)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor trims String input params of all methods of class marked by @Trimmed if " +
+            "trimming is enabled by @EnableStringTrimming ")
+    void trimmedAnnotationPostProcessor_trimmedClass_inputParams(@Autowired TrimmedService service) {
+        String inputArgs = " Simba Bimba  ";
+        String actual = service.getTheSameString(inputArgs);
+
+        assertEquals(inputArgs.trim(), actual);
+    }
+
+    @Test
+    @Order(13)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor trims String return values of all methods of class marked by @Trimmed if " +
+            "trimming is enabled by @EnableStringTrimming ")
+    void trimmedAnnotationPostProcessor_trimmedClass_returnValue(@Autowired TrimmedService service) {
+        String inputArgs = "Simba Bimba";
+        String actual = service.wrapStringWithSpaces(inputArgs);
+
+        assertEquals(inputArgs, actual);
+    }
+
+    @Test
+    @Order(14)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor not trims String input args methods of class that not marked by @Trimmed")
+    void trimmedAnnotationPostProcessor_notTrimmedClass_inputParams(@Autowired NotTrimmedService service) {
+        String inputArgs = " Simba Bimba ";
+        String actual = service.getTheSameString(inputArgs);
+
+        assertEquals(inputArgs, actual);
+    }
+
+    @Test
+    @Order(15)
+    @DisplayName("TrimmedAnnotationBeanPostProcessor not trims String input args methods of class that not marked by @Trimmed")
```

Comment:
@Stanislav-Zabramnyi let's test if method params (not classes) are marked with `@Trimmed`